chromium_src/content/browser/browser_context.cc

```diff
@@ -0,0 +1,41 @@
+/* Copyright (c) 2020 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "content/public/browser/browser_context.h"
+
+#include <string>
+
+#include "base/memory/ref_counted.h"
+#include "content/browser/dom_storage/dom_storage_context_wrapper.h"
+#include "content/browser/dom_storage/session_storage_namespace_impl.h"
+#include "content/browser/renderer_host/render_view_host_delegate.h"
+#include "content/public/browser/render_view_host.h"
+#include "content/public/browser/session_storage_namespace.h"
+#include "content/public/browser/storage_partition.h"
+#include "content/public/browser/web_contents.h"
+
+namespace content {
+
+scoped_refptr<content::SessionStorageNamespace> CreateSessionStorageNamespace(
+    content::StoragePartition* partition,
+    const std::string& namespace_id) {
+  content::DOMStorageContextWrapper* context_wrapper =
+      static_cast<content::DOMStorageContextWrapper*>(
+          partition->GetDOMStorageContext());
+
+  return content::SessionStorageNamespaceImpl::Create(context_wrapper,
+                                                      namespace_id);
```

Comment:
There is an interesting quirk, saying that `namespace_id` length is basically hardcoded [1]. Wondering if this could potentially hit us? If we are fine it's worth mentioning in a comment

[1] https://source.chromium.org/chromium/chromium/src/+/master:third_party/blink/common/dom_storage/session_storage_namespace_id.cc;l=17?q=session_storage_namespace_id

Reply:
Oh, nice find! I will modify the code to use and MD5 hash and pad the hash in a way that it cannot collide with GUID generation.